subprojects/gstreamer/plugins/tracers/gstframerate.c

```diff
@@ -134,9 +355,23 @@ gst_framerate_tracer_display_current_fps (GstFramerateTracer *self,
   if (dr == 0.0) {
     gst_tracer_record_log (tr_framerate, self->pid, current_ts, frames_rendered,
         frames_dropped, average_fps, -1.0f, -1.0f);
+    g_snprintf (data, sizeof (data), CLIENT_SOCKET_DATA_FORMAT,
+        self->pid, current_ts, frames_rendered, frames_dropped, average_fps,
+        -1.0f, -1.0f);
   } else {
     gst_tracer_record_log (tr_framerate, self->pid, current_ts, frames_rendered,
         frames_dropped, -1.0f, rr, dr);
+    g_snprintf (data, sizeof (data), CLIENT_SOCKET_DATA_FORMAT,
+        self->pid, current_ts, frames_rendered, frames_dropped, -1.0f, rr, dr);
+  }
+
+  if (self->unix_socket_path != NULL) {
+    send_data = g_new0 (GstFramerateClientSocketSendData, 1);
+    send_data->self = self;
```

Comment:
there's a possible race condition: element destruction vs thread that reports fps. We should hold a ref here
```c
send_data->self = gst_object_ref (self);
```